### ISSUE TITLE
Feature Request: Weather

### ISSUE THREAD

**Issue by sb76117** [theothernt/AerialViews#36]
I know this is probably a big ask, but is there any way to have weather info on the opposite side of the clock/location info?

Wishing you the best of luck in the new year and thank you for this app!

**Comment by theothernt** [comment]
That's a good request and something I will look into. The only issue will be where to get the weather from, and will it be free or will a user have to sign up for a service and get their own 'key' to use the data.

**Comment by sb76117** [comment]
A user supplied API key would probably be the best way to go to avoid call rate-limiting. I use [this Kodi add-on](https://forum.kodi.tv/showthread.php?tid=352028) that has options to use a number of different weather services.

**Comment by FaberfoX** [comment]
The default "backdrop" screensaver has the weather as an option, so (and I'm guessing here) maybe there's a way for you to get it from it, using reflection? I did that a long time ago on a now abandoned app to get some internals that were not in an exposed api.

**Comment by theothernt** [comment]
> The default "backdrop" screensaver has the weather as an option, so (and I'm guessing here) maybe there's a way for you to get it from it, using reflection? I did that a long time ago on a now abandoned app to get some internals that were not in an exposed api.

That's a good idea, I'll probably look into it although my main worry would be that the Play Store won't allow it. And this app does have a non-store version but I'd like to keep both the same, where possible.

Also, so far, library support for other weather APIs seems pretty good, which mean I'd have less work to do! 😅

**Comment by yknowwhatudid** [comment]
Is there any update to this suggestion ?

**Comment by theothernt** [comment]
> Is there any update to this suggestion ?

Yes, the next version of the app is 1.5.1 - I'm hoping to have the weather overlay working in 1.5.2 but it could be a couple of months away.

I will post a beta here when it's ready for testing.

**Comment by jhoblitt** [comment]
It sounds like people are asking for functionality which overlaps with magic mirror plugins and layout.  I wonder how hard it would be support rtsp/etc. streams in this app? That would allow ffmpeg/etc. to be used to setup a magic mirror feed.

**Comment by theothernt** [comment]
> It sounds like people are asking for functionality which overlaps with magic mirror plugins and layout. I wonder how hard it would be support rtsp/etc. streams in this app? That would allow ffmpeg/etc. to be used to setup a magic mirror feed.

It sounds similar at a high level, but they're quite different. In reality, Aerial Views has no plugin system - just a fixed set of options to show or hide certain text. 

I've only recently started calling them 'overlays' as it matches with the Mac version of this screensaver :)

**Comment by theothernt** [comment]
Just a small update. Aerial Views 1.6 has been released with the new slot/overlays system so I'm hoping to have a weather overlay in the next release (if all goes well!)

**Comment by skorpioskorpio** [comment]
Also interested in this, any update?

**Comment by skorpioskorpio** [comment]
Understood, I imagine it's fairly complicated to pull off. I am fairly sure I'm not the only one that ran across this screensaver while looking for another, a specific other, that happens to have that feature (not that it works very reliably). 

If I may offer a few suggestions on what to display from Open Weather Map as I use that data myself in Home Assistant. You currently have 2 lines per corner built in, it would be incredibly cool to have fields for; "location", "current" (with temp, conditions (maybe as an icon representation), humidity and wind speed format options), "tomorrow" or "tonight" (depending on where you are in the day cycle, same format as "current"), as well as "a 5 day header" and "5 day data" (temp and icon only). Dynamic would be incredible, being able to rotate between those options, basically being able to pop or slide pairs in and out of one of the corners so that you could essentially display all of those but only a pair at a time, or do that for any data for that matter as that seems like it would be a useful feature to "Now Playing". 

Anyway just some thoughts, this is a really cool app as it is. I personally intend to use this as a baseline for what gets displayed on a TV when I walk into a room (on a Fire TV). So I plan on using it more as an information display than a screensaver, where it is displayed (or not) based on Home Assistant automatons. But so far, for that purpose, it's awesome, thank you!

**Comment by theothernt** [comment]
> Understood, I imagine it's fairly complicated to pull off. I am fairly sure I'm not the only one that ran across this screensaver while looking for another, a specific other, that happens to have that feature (not that it works very reliably).

Compared to most of the app, it is. There are lots of possible issues with requesting the weather, not to mention API limits. And obviously smooth video playback and stability are most important.

As for certain other apps, I've been told my SMB/Samba works reliably too!

> If I may offer a few suggestions on what to display from Open Weather Map as I use that data myself in Home Assistant. You currently have 2 lines per corner built in, it would be incredibly cool to have fields for; "location", "current" (with temp, conditions (maybe as an icon representation), humidity and wind speed format options), "tomorrow" or "tonight" (depending on where you are in the day cycle, same format as "current"), as well as "a 5 day header" and "5 day data" (temp and icon only). 

So, initially it'll be very simple like what you see on a Google Home/Nest - ie. weather icon, temp. Assuming that goes well and is stable, I can add more features. 

I hope to have: current weather, 1 day, 3 day, 5 day forecasts with min/max temp. The idea is to have maybe 3 or 4 weather overlays you can customize as I have a ton of data like a) sunrise/sunset times b) humidity c) wind/gust speed/direction d) feels-like temp etc.

I have been given a permission to use OpenWeatherMap and I might add Open Meteo as an option or backup.

> Dynamic would be incredible, being able to rotate between those options, basically being able to pop or slide pairs in and out of one of the corners so that you could essentially display all of those but only a pair at a time, or do that for any data for that matter as that seems like it would be a useful feature to "Now Playing".

For the moment, with the exception of the weather status icons, weather and now playing will just be simple text. I hope to make it a bit fancier in future, but we'll see.

> Anyway just some thoughts, this is a really cool app as it is. I personally intend to use this as a baseline for what gets displayed on a TV when I walk into a room (on a Fire TV). So I plan on using it more as an information display than a screensaver, where it is displayed (or not) based on Home Assistant automatons. But so far, for that purpose, it's awesome, thank you!

Thanks for saying that. It's always nice to hear how people use the screensaver, from just a single TV, to many TVs or on a tabled used in home automation... or just on a phone!

**Comment by ToddCrimson** [comment]
hello @theothernt , wondering if this is completed and weather support is now in the screensaver? I see that https://github.com/theothernt/AerialViews/issues/155 is closed as completed.

If completed, is there a method to turn it on? I played around in settings, didn't see it. 

Thank you

**Comment by theothernt** [comment]
> wondering if this is completed and weather support is now in the screensaver? I see that [#155](https://github.com/theothernt/AerialViews/issues/155) is closed as completed.

I'm afraid the feature hasn't been completed yet. I do hope to have an update in the next month or two.

**Comment by ToddCrimson** [comment]
Thank you, appreciate all your efforts on this awesome app.
PS: Do you have any recommendations (since you are close to this) for an alternative screensaver management app on Android TV. I ask as I used LineageOS on my Nvidia Shield Pro. 
The app bundled with LineageOS which is responsible for managing screensavers is buggy.
When 5 (or 10, or 15) mins are up, the screensaver doesn't come on. Sometimes it will just show blank screen.
This happens whether you choose their default screen saver or aerial. 
I tested aerial on stock OS, and it works.
It also works on LineageOS if you test it, by clicking start now.

Logs show that there is some fatal exception to this specific app.

So question for you: Do you have any idea for an alternative app that I can use for setting the screensaver time and making it come on?

**Comment by theothernt** [comment]
> Do you have any recommendations (since you are close to this) for an alternative screensaver management app on Android TV. I ask as I used LineageOS on my Nvidia Shield Pro.

If you wouldn't mind, I'll continue with this separate issue over on [this thread started by another user who is also running LineageOS](https://github.com/theothernt/AerialViews/issues/195) on a tablet.

**Comment by crowley88** [comment]
I'd love to test it for you as it's definitely a feature I would like to use

**Comment by theothernt** [comment]
> I'd love to test it for you as it's definitely a feature I would like to use

Sign up here... https://play.google.com/apps/testing/com.neilturner.aerialviews

You can visit the same link to opt-out of the betas at a later date.

**Comment by crowley88** [comment]
Just tested and works perfect for Nvidia shield not tried all locations but tested a few and it's perfect for me. Thanks for the feature it makes my screensaver look even better

**Comment by theothernt** [comment]
Thanks, there will be more weather info plus customisation in future. I may tweak the icons a bit as well.

**Comment by theothernt** [comment]
v1.7.8 is out now on GitHub with the Weather overlay. Expect the Play Store version in a few days.
